azurerm/resource_arm_route_test.go

```diff
@@ -79,6 +79,34 @@ func TestAccAzureRMRoute_multipleRoutes(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMRoute_removed(t *testing.T) {
+	ri := acctest.RandInt()
+	location := testLocation()
+	preConfig := testAccAzureRMRoute_basic(ri, location)
+	postConfig := testAccAzureRMRoute_basicUpdate(ri, location)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMRouteDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: preConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMRouteExists("azurerm_route.test"),
+				),
+			},
+
+			{
+				Config: postConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMRouteExists("azurerm_route.test1"),
```

Comment:
I believe this is failing since this route isn't in the `postConfig`?

Reply:
@paultyng Sorry I never actually fixed this test to fail in the proper way.  I had accidentally added it to this PR.  Let me update the test to cause the race condition I am describing properly and I will push a change shortly

Reply:
Thanks, should the KeyVault tests fail as well as written? Running them now to see the race issue.

Reply:
The seem to have passed for me.

Reply:
I believe I currently have it in a working state...but it is very very easy to break my test again. 

in the 'testAccAzureRMKeyVaultAccessPolicy_basic' function remove the following line from that tags block
```
    policy_object_id = "${data.azurerm_client_config.current.service_principal_object_id}"
```

This should break it in the same way I was experiencing before

Reply:
Btw I just updated the PR to cause this test to fail due to this race condition. Should be failing now.